docs/internals/contributing/writing-documentation.txt

```diff
@@ -341,6 +341,39 @@ documentation:
     Five
     ^^^^
 
+* Add headers for functions, classes, and methods in reference documentation:
+
+  - Each function, class, method, or attribute in reference documentation should
+    have a header using the appropriate heading level based on document hierarchy.
+  - Use the exact name of the item as the header text.
+  - Place the header directly above the directive documentation.
+  - For better accessibility and navigation, ensure these headers are
+    consistently applied across all reference documentation.
```

Comment:
```suggestion
```

I don't personally feel this line is necessary. People who are writing documentation should not need to ensure that the headers are consistently applied elsewhere - it should already be so.